engines/sherlock/sound.cpp

```diff
@@ -0,0 +1,236 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include "sherlock/sherlock.h"
+#include "sherlock/sound.h"
+#include "common/config-manager.h"
+#include "audio/audiostream.h"
+#include "common/algorithm.h"
+#include "audio/mixer.h"
+#include "audio/decoders/raw.h"
+
+namespace Sherlock {
+
+Sound::Sound(SherlockEngine *vm, Audio::Mixer *mixer): _vm(vm), _mixer(mixer) {
+	_digitized = false;
+	_music = false;
+	_voices = 0;
+	_playingEpilogue = false;
+	_diskSoundPlaying = false;
+	_soundPlaying = false;
+	_soundIsOn = &_soundPlaying;
+	_curPriority = 0;
+
+	_soundOn = true;
+	_musicOn = true;
+	_speechOn = true;
+
+	_vm->_res->addToCache("MUSIC.LIB");
+	_vm->_res->addToCache("TITLE.SND");
+	_vm->_res->addToCache("EPILOGUE.SND");
+	_vm->_res->addToCache("SND.SND");
+}
+
+/**
+ * Saves sound-related settings
+ */
+void Sound::syncSoundSettings() {
+	_digitized = !ConfMan.getBool("mute");
+	_music = !ConfMan.getBool("mute") && !ConfMan.getBool("music_mute");
+	_voices = !ConfMan.getBool("mute") && !ConfMan.getBool("speech_mute") ? 1 : 0;
+}
+
+void Sound::loadSound(const Common::String &name, int priority) {
+	// No implementation required in ScummVM
+}
+
+char Sound::decodeSample(char sample, byte& prediction, int& step) {
+	char diff = ((sample & 0x07) << step);
+
+	if (sample & 0x08) {
+		if (prediction > diff)
+			prediction = prediction - ((sample & 0x07) << step);
+		else
+			prediction = 0;
+	} else {
+		if (prediction < 0xff - diff)
+			prediction = prediction + ((sample&0x07) << step);
+		else
+			prediction = 0xff;
+	}
+
+
+	if ((sample & 0x07) >= 5 && step < 3) {
+		step ++;
+	} else if ((sample & 0x07) == 0 && step > 0) {
+		step --;
+	}
+
+	return prediction;
+}
+
+bool Sound::playSound(const Common::String &name, WaitType waitType, int priority) {
+	stopSound();
```

Comment:
Something seems to be wrong here, because the speech in the intro is distorted for me. I'm mentioning this mainly in case it's a regression.